### PR TITLE
Adding Namespace Capacity Terraform Resource

### PR DESCRIPTION
**Pull request by pauloh-temporal**
## Checklist

1. Closes CLDGR-751

2. How was this tested:

3. Any docs updates needed?

---

> [!NOTE]
> Add capacity configuration to `temporalcloud_namespace`, with update support (provisioned/on_demand), docs/tests, and dependency updates; creation with capacity is currently disallowed.
> 
> - **Resource: `internal/provider/namespace_resource.go`**
>   - Add optional `capacity` attribute (mode: `provisioned`/`on_demand`, value when provisioned) to schema/model/state.
>   - Implement parsing via `getCapacityFromModel` and propagate to `NamespaceSpec.CapacitySpec` on update.
>   - Disallow setting `capacity` during create (returns error).
>   - Sync capacity from API spec into state.
> - **Tests: `internal/provider/namespace_resource_test.go`**
>   - Add acceptance tests for capacity transitions (null → provisioned → on_demand) with polling for completion.
> - **Docs: `docs/resources/namespace.md`**
>   - Document new `capacity` attribute and nested schema.
> - **Dependencies: `go.mod`, `go.sum`**
>   - Update Temporal SDK/cloud-sdk/api and add testing deps (e.g., `testify`), minor gRPC/protobuf adjustments.
> - **Removals**
>   - Delete obsolete tests/helpers: `internal/provider/metrics_endpoint_resource_test.go`, `internal/provider/testing.go`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 68909259f2d6a5e6721c5929a0991c77456567d9. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: internal/client/client.go [hunk 1]
```diff
@@ -112,3 +116,69 @@ func AwaitAsyncOperation(ctx context.Context, cloudclient *Client, op *operation
 		}
 	}
 }
+
+func AwaitForFulfillment(ctx context.Context, cloudclient *Client, n *namespace.Namespace, predicate ...AwaitPredicate) (bool, error) {
```
Comment (anekkanti) [inline internal/client/client.go:120]: Unused?

File: internal/client/client.go [hunk 2]
```diff
@@ -112,3 +116,69 @@ func AwaitAsyncOperation(ctx context.Context, cloudclient *Client, op *operation
...
+		b, err := p(ctx, cloudclient, n)
+		if err != nil {
+			errs = multierror.Append(errs, err)
+		}
+		bValue = bValue && b
+	}
+	return bValue, errs
+}
+
+func AwaitNamespaceCapacityOperation(ctx context.Context, cloudclient *Client, n *namespace.Namespace) (bool, error) {
```
Comment (anekkanti) [inline internal/client/client.go:136]: Unused?

File: internal/provider/namespace_resource_test.go [hunk 3]
```diff
@@ -791,3 +793,141 @@ PEM
...
+							} else {
+								// success
+								return nil
+							}
+						}
+					}
+					return fmt.Errorf("timed out waiting for capacity change")
+				},
+			},
+			{
```
Comment (anekkanti) [inline internal/provider/namespace_resource_test.go:924]:
```suggestion
			{
			    // Rever namespace back to defaults
				Config: config(name, "null"),
			},
			{
```
